modules/imap/js_modules/route_handlers.js

```diff
@@ -9,20 +10,78 @@ function applyImapMessageListPageHandlers(routeParams) {
 
     Hm_Message_List.set_row_events();
 
-    $('.core_msg_control').on("click", function(e) {
+    $('.core_msg_control').on("click", function (e) {
         e.preventDefault();
-        Hm_Message_List.message_action($(this).data('action')); 
+        Hm_Message_List.message_action($(this).data('action'));
     });
-    $('.toggle_link').on("click", function(e) {
+
+    $('.toggle_link').on("click", function (e) {
         e.preventDefault();
         Hm_Message_List.toggle_rows();
     });
 
+    if (listPath.substr(0, 4) === 'imap') {
+        setup_imap_folder_page(listPath);
+    } else if (listPath === 'combined_inbox') {
+        setup_imap_message_list_content_page();
+    }
```

Comment:
Why are you doing this?

Reply:
The if condition (listPath.substr(0, 4) === 'imap') allows IMAP folders to be treated separately, ensuring that messages are displayed correctly for each folder
and The else if (listPath === 'combined_inbox') condition ensures that when the user navigates to an inbox that combines messages from multiple folders, the code calls a suitable function to display that view.

Reply:
Aren't messages correctly displayed currently for each folder?

Reply:
yes this is the case for the Scheduled folder

Reply:
`else if (listPath === 'combined_inbox') {setup_imap_message_list_content_page();}`
I just deleted this part, a thinking error on my part.